### DISCUSSION TITLE
Customize setting title

### DISCUSSION THREAD

**Discussion by alexweininger** [microsoft/vscode-discussions#406] [Extension Development QnA] [answered]
We've recently changed our extension display name, and want to reflect this change in the settings editor. However, doing so will require us to change the setting ids, which we want to avoid.

As described by the [contributes.configuration docs](https://code.visualstudio.com/api/references/contribution-points#Configuration-schema), the setting title is generated from the setting key.

> <img width="758" alt="image" src="https://user-images.githubusercontent.com/12476526/211948935-6200dc4e-8746-4888-a35a-6a2d59b791ed.png">

Our setting keys are prefixed with the extension id, which no longer matches the extension display name. So now the setting title does not match the extension display name, which is confusing for users.

We'd like to be able to change the setting title without changing the setting key, since we'd have to migrate/deprecate all of our settings.

Possible solutions:
1. Allow an optional property to override the value used to generate the setting title
2. Prefix the setting title with the extension display name instead of the extension id by default

---
cc: @isidorn 
Related issue: https://github.com/microsoft/vscode-azureresourcegroups/issues/447

**Comment by gjsjohnmurray** [comment, marked as answer]
I think https://github.com/microsoft/vscode/issues/150891 will address this.

**Reply by isidorn** [reply]
@gjsjohnmurray thanks a lot! I was not aware of that one.
@alexweininger I just commented on that issue. Maybe you can as well and we see how the situation unfolds.
